lib/ocpp/v201/charge_point.cpp

```diff
@@ -3805,6 +3836,36 @@ void ChargePoint::execute_change_availability_request(ChangeAvailabilityRequest
     }
 }
 
+// K01.27 - load profiles from database
+void ChargePoint::load_charging_profiles() {
+    try {
```

Comment:
What do you have in mind?

Reply:
You could make this function return a bool and then clear the database when you get `false` on the call for example. There are probably more elegant solutions to deal with this but this at least ensures you can continue after.

It's still a big issue if it happens at all since it means we can corrupt the database in some way and should find out what causes that.

Reply:
I slightly refactored the try/catch removing two cases. I am not sure about the idea of fully deleting the database; it seems a bit too aggressive. Currently, there are three ways to fail. This is ignoring the idea that the DB is corrupted.

The only way I see for any of these to happen is if the query is broken. We currently have test which should ensure this does not occur. In the case of bad data making it into the DB then failures would be a good indication of a system issue.

Load Failure - A call to get all profiles by the evse is done. If this query fails we could just let the system run and the CSMS could resend the data. The other option would be to not boot.

Add Failure - During boot the system validates profiles and adds them back to the in-memory cache. There is a call to add these to the database, which I find odd, but if this fails it does not matter. The profile is already present.

Delete Failure - This is the only case that would cause divergence in the two states. The in-memory database would not have something the database does. Since all processing is on the in-memory cache nothing critical should fail.

Reply:
The comment on database corruption was a bit misleading. The system would have to handle this at a higher level. Specifically, within the `ChargePoint` constructor. Currently, it appears that if an error is throw the system would not boot which is the correct behavior.

Overall, for this method the goal would be [this issue](https://github.com/EVerest/libocpp/issues/713). In this case we would not have to worry about loading any data back into memory and the bulk of this would be fairly minimized. I would highly suggest community buy in on this.

I can add a diagram to the mentioned issue but the flow would be:

1. Boot the system 
  - If a connection cannot be made to the DB, in the constructor, do not boot.
2. Load `ChargingProfiles` for validation
  - `ChargingProfiles` can be invalid if, for example, an EVSE was removed.
  - If the `ChargingProfiles` cannot be loaded do not boot.
3. Delete invalid `ChargingProfiles`
  - If a delete cannot be completed do not boot

In any "do not boot" case we would need to inform the CSMS of the issue.

Reply:
Agree that we need community buy-in, just wanted to make sure that the current thoughts were being tracked in the longer term discussion.

Reply:
Currently we don't check the entire database at boot for corruption. This might be a nice addition for the future but I would certainly exclude that from the change of removing the runtime copy of the data and only using the database for storage.

In any case I would also argue that if a delete is not possible we should continue instead of not booting, and still somehow report on this to the CSMS indeed. If we do not boot libocpp we are not even able to contact the CSMS since that is our way of contact.

For now I will resolve this conversation since most of these issues will be solved by #713